cocotb/decorators.py

```diff
@@ -85,13 +86,8 @@ class RunningTask(object):
         task.kill() will destroy a coroutine instance (and cause any Join
         triggers to fire.
     """
-    def __init__(self, inst):
-
-        if inspect.iscoroutine(inst):
-            self._natively_awaitable = True
-        elif inspect.isgenerator(inst):
-            self._natively_awaitable = False
-        else:
+    def __init__(self, inst: collections.abc.Coroutine):
+        if not isinstance(inst, collections.abc.Coroutine):
```

Comment:
```suggestion
        if not inspect.isawaitable(inst):
```
https://docs.python.org/3.5/library/inspect.html#inspect.isawaitable

Reply:
That's not strict enough, because we need the object to have `send`, `throw`, and `close` methods.